core/controllers/collection_editor.py

```diff
@@ -28,6 +28,28 @@
 current_user_services = models.Registry.import_current_user_services()
 
 
+def get_matching_exploration_dicts(query_string, search_cursor):
+    """Given a query string and a search cursor, returns a list of exploration
+    dicts that satisfy the search query.
+    """
+
+    exp_ids, new_search_cursor = (
+        exp_services.get_exploration_ids_matching_query(
+            query_string, cursor=search_cursor))
+    exploration_list = []
+
+    exploration_list = (
```

Comment:
Maybe call this collection_node_metadata_dicts

(I'm trying to find a name that's not 'summary' for the tiny dicts, so we don't get confused between this and the standard exploration summaries.)